pandas/_libs/groupby.pyx

```diff
@@ -1075,6 +1075,8 @@ def group_mean(
                     y = val - compensation[lab, j]
                     t = sumx[lab, j] + y
                     compensation[lab, j] = t - sumx[lab, j] - y
+                    if compensation[lab, j] != compensation[lab, j]:
```

Comment:
this is to check for nan?  can you use an explicit check?  If not, please add a comment about what you're doing and why

Reply:
Hi @jbrockmendel, I tried do it explicitly but it is not working.  I tried to use `utils.is_nan` since it is `no gill` it is not compiling. So I just used this way to determine Nan (That's what we are doing in `utils.is_nan`  as well.)
I will add appropriate comment in the code